### PR TITLE
Enable Windows CI on Azure Pipelines #9918

### PR DESCRIPTION
**Pull request by mlline00**
PR to track progress toward running Windows test suite on Azure Pipelines.  Opening the PR early to give visibility into the progress. I expect a few iterations before merge.

### CODE REVIEW

**Comment by mlline00** [comment]
There is one flaky py37 win-32 test:

tests/test_activate.py::ShellWrapperIntegrationTests::test_powershell_PATH_management

Root cause is the infamous SSLError("Can't connect to HTTPS URL because the SSL module is not available") when attempting to connect to https://repo.anaconda.org. I'm not sure what to do about it because the recommended fix to activate the conda environment, but this test executes two "conda deactivate" commands. The test does occasionally pass, but I'm not able to reliably reproduce a passing scenario.

Searching historically, this test was originally xfailed on python 3 because OpenSSL was not on the path (commit fb233e3288cf4f0eb4eb2eeed8f27039b1e97491) , but the xfail was removed a few days later without meaningful commentary (commit 11d66a8ba78c38d079011996ff49f8be1c20586f).   Pure speculation, but AppVeyor is only configured to test win-64 python and the test consistently passes on win-64 platforms, thus the xfail would be inappropriate and have to be removed.

To summarize, I don't see a clear path to resolve this test.

**Comment by mcg1969** [comment]
@mlline00 I actually just got AppVeyor under its timeouts by breaking apart the individual tests into several pieces.

At the same time, I've had great results with other repos (conda-pack, anaconda-project, nb_conda_kernels) using GitHub Actions, an evolution of Azure Pipelines. I think it would be great to standardize on it.

**Comment by mlline00** [comment]
@mcg1969 Great to see the test suite consistently passing again! I'm (largely) indifferent to the CI platform. Went down the Azure Pipelines path based on the feedback in #9918.

This PR makes two material changes to the testing:
- Replaces python 3.6 with 3.7
- Covers 32 and 64-bit platforms for each version of python

Additionally, there was a 64-bit python 2.7 bug uncovered and fixed in conda/common/_logic.py (see comment for d1aecbd).

It won't hurt my feelings if this PR is closed without merging.  Otherwise, let me know how you'd like to proceed.

**Comment by mcg1969** [comment]
@mlline00 would you be willing to cherry pick https://github.com/conda/conda/commit/d1aecbdc98537d36f6de9f664606360a1414aa5c into an independent PR? We should definitely merge that immediately.
I feel like I should defer to the rest of the team on the overall CI effort, although I must admit that if we were to make a change, it's my view we should go to GitHub actions if simply in anticipation of better support and integration from Microsoft on this moving fowrard.

**Comment by mlline00** [comment]
@mcg1969 Cherry picked d1aecbd and submitted as PR #10108
